src/Misc/indexedset.jl

```diff
@@ -0,0 +1,92 @@
+"""
+    IndexedSet{T}
+
+A hybrid data structure combining properties of both a vector and a set.
+
+By implementation you can push things to its end and iterate it sequentially;
+but duplicates are simply dropped (like in a set). Therefore, given any element you can
+test in O(1) if it is contained in the set, and find out at which position.
+
+This structure is suited well for applications such as orbit algorithms, 
+where efficient membership checks and indexed retrieval are beneficial.
+
+```jldoctest
+julia> s = IndexedSet(["ball", "flower", "house"])
+Indexed set ["ball", "flower", "house"]
+
+julia> "ball" in s
+true
+
+julia> s("flower")   # get the position
+2
+
+julia> s[2]
+"flower"
+
+julia> "stone" in s
+false
+
+julia> s("stone")    # the "position" of objects not in there is 0
+0
+
+julia> push!(s, "ball", "stone")    # only "new" things are actually added
+Indexed set ["ball", "flower", "house", "stone"]
+```
+"""
+struct IndexedSet{T}
+  data::Vector{T}
+  pos::Dict{T,Int}
+
+  IndexedSet{T}() where {T} = new{T}(Vector{T}(), Dict{T,Int}())
+end
+
+function IndexedSet(v::Vector{T}) where {T}
+  s = IndexedSet{T}()
+  for x in v
+    push!(s, x)
+  end
+  return s
+end
+
+function Base.show(io::IO, s::IndexedSet)
+  if is_terse(io) || get(io, :typeinfo, nothing) == typeof(s)
+    io = IOContext(io, :typeinfo => typeof(s.data))
+    print(io, s.data)
+  else
+    print(io, "Indexed set ", s.data)
```

Comment:
```suggestion
    print(io, "IndexedSet(", s.data, ")")
```
IMO it makes sence to have this print with parentheses, as e.g. `Base.Set` does as well. This then has the side-effect that a simple index set like `IndexedSet(["ball", "flower", "house"])` prints exactly like it is constructed

Reply:
I agree, I've now changed this in the recent commit.